instrumentation/spring/spring-webmvc/spring-webmvc-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v6_0/DispatcherServletInstrumentation.java

```diff
@@ -61,13 +62,26 @@ public static class HandlerMappingAdvice {
     public static void afterRefresh(
         @Advice.Argument(0) ApplicationContext springCtx,
         @Advice.FieldValue("handlerMappings") List<HandlerMapping> handlerMappings) {
-      if (springCtx.containsBean("otelAutoDispatcherFilter")) {
-        OpenTelemetryHandlerMappingFilter filter =
-            (OpenTelemetryHandlerMappingFilter) springCtx.getBean("otelAutoDispatcherFilter");
-        if (handlerMappings != null) {
-          filter.setHandlerMappings(handlerMappings);
+
+      if (handlerMappings == null || !springCtx.containsBean("otelAutoDispatcherFilter")) {
+        return;
+      }
+
+      Object bean = springCtx.getBean("otelAutoDispatcherFilter");
+      OpenTelemetryHandlerMappingFilter filter;
+      if (bean instanceof OpenTelemetryHandlerMappingFilter) {
+        // inline advice: no proxy class is used
+        filter = (OpenTelemetryHandlerMappingFilter) bean;
+      } else {
```

Comment:
Imo instead of doing the reflection here there should be an API to do this. Something that would unwrap the object when it is proxy and return the input when it is not. So you could write `OpenTelemetryHandlerMappingFilter filter = = (OpenTelemetryHandlerMappingFilter) unwrap(springCtx.getBean("otelAutoDispatcherFilter"));` Instead of making the delegate field public I'd probably try to make the proxy instance implement an interface that allow unwrapping. If needed that interface could be hidden from reflection https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/647fc45914725b4805f63b97c0084f951397c774/instrumentation/internal/internal-reflection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/reflection/ReflectionHelper.java#L59 I guess using the reflection as currently would also be fine for start.

Reply:
Hi, I've managed to implement this by introducing an `IndyProxy` interface that allows us to easily unwrap the proxies when needed with an `IndyProxyHelper` class. It took me a while and it's definitely a bit more complicated than a public field, but it's way more elegant to use. Also it was a very good way to learn about "how to hide things from reflection API" :-).

Let me know if that's close to what you had in mind here. One thing that is still missing though is properly testing hiding proxy method and interface in the reflection instrumentation.

Reply:
Update: tests have been added so it should be ready for review now.